### PR TITLE
Fix CI for thumb* targets

### PR DESCRIPTION
**Pull request by paoloteti**
Main fix is inside `utest` (implement Termination to avoid ICE) all the rest are minor fixes due to the new `testcrate`.

Thumb6m needs some extra fixes: some undefined reference and a panic raised removing assembly implementation not in the list.

Still allow failures on TravisCI and keep my `utest` fork for a while. Let's see if is stable.

### CODE REVIEW

File: testcrate/Cargo.toml [hunk 1]
```diff
@@ -14,12 +14,15 @@ rand = { version = "0.4", features = ["i128_support"] }
...
-features = ["mangled-names", "no-lang-items"]
+features = ["no-lang-items"]
 
 [target.'cfg(all(target_arch = "arm", not(any(target_env = "gnu", target_env = "musl")), target_os = "linux"))'.dev-dependencies]
-test = { git = "https://github.com/japaric/utest" }
-utest-cortex-m-qemu = { default-features = false, git = "https://github.com/japaric/utest" }
-utest-macros = { git = "https://github.com/japaric/utest" }
+test = { git = "https://github.com/paoloteti/utest", branch = "ci" }
+utest-cortex-m-qemu = { default-features = false, git = "https://github.com/paoloteti/utest", branch = "ci" }
+utest-macros = { git = "https://github.com/paoloteti/utest", branch = "ci" }
```
Comment (alexcrichton) [inline testcrate/Cargo.toml]: For tracking purposes, are the changes here available as upstream PRs perhaps?

File: testcrate/Cargo.toml [hunk 2]
```diff
@@ -14,12 +14,15 @@ rand = { version = "0.4", features = ["i128_support"] }
...
 
 [features]
 c = ["compiler_builtins/c"]
+mem = ["compiler_builtins/mem"]
+mangled-names = ["compiler_builtins/mangled-names"]
```
Comment (alexcrichton) [inline testcrate/Cargo.toml]: Could this perhaps be turned on by default? I was hoping to keep `cargo test` working in the `testcrate` directory "mostly working by default" if possible

File: testcrate/src/lib.rs [hunk 3]
```diff
@@ -1,3 +1,5 @@
+#![cfg_attr(feature = "no_std", no_std)]
```
Comment (alexcrichton) [inline testcrate/src/lib.rs]: Could this unconditionally be `#![no_std]`?

Comment (alexcrichton) [reply testcrate/src/lib.rs]: (or this entire file could be emptied)

File: testcrate/Cargo.toml (hunk 1 above)
Comment (paoloteti) [reply testcrate/Cargo.toml]: Yes it is upstream now, I'll update the URL

File: testcrate/Cargo.toml (hunk 2 above)
Comment (paoloteti) [reply testcrate/Cargo.toml]: Do you mean 'default = ["mangled-names"]' only ?

**Comment by paoloteti** [comment]
Recent `rust-lang/rust` PR [48143](https://github.com/rust-lang/rust/pull/48143) broke again `utest`. 

If accepted on `utest` upstream PR [2](https://github.com/japaric/utest/pull/2) will fix the issue, so let's wait.

File: testcrate/Cargo.toml (hunk 2 above)
Comment (alexcrichton) [reply testcrate/Cargo.toml]: Indeed!

**Comment by alexcrichton** [comment]
Ok! Regardless though this all looks good to me, so let's land this and then we can edit .travis.yml to start blocking again in a follow-up?

@bors: r+

**Pull request merged**